doc/1-intro/glossary.md

```diff
@@ -0,0 +1,145 @@
+# glossary of common terms
+
+**2-op**, **3-op**, **4-op**...: the number of FM operators used to generate a sound. more operators allow for more complex sounds.
+
+**ADPCM**: adaptive differential pulse code modulation. this is a variety of DPCM with a more complex method of storing the amplitude differences.
+
+**ADSR**: attack, decay, sustain, release. these are the four necessary values for a basic volume envelope.
+
+**algorithm**: the way in which the operators in an FM instrument interact.
+- when two operators connect to the same point, their sounds are added together.
+- when two operators are connected left to right, the left is the modulator and the right is the carrier sound that is modified.
+
+**bitbang**: to achieve PCM sound by sending a rapid stream of volume commands to a non-PCM channel.
+
+**BRR**: a type of lossy ADPCM used by the SNES. it has a fixed compression ratio; groups of 32 bytes (16 samples) are encoded in 9 bytes each.
```

Comment:
# BRR is not ADPCM.

It actually uses LPC. It compresses blocks of 16 samples into 4-bit coded samples with filter and range parameters.